### PR TITLE
mesa: patch for dri2_query_dma_buf_modifiers

### PR DESCRIPTION
**Pull request by balav08**
Signed-off-by: balav08 <balaji.velmurugan@ltts.com>

**- What I did**

**- How I did it**

### CODE REVIEW

File: recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ dri2_from_fds(__DRIscreen *screen, int width, int height, int fourcc,
+               int *fds, int num_fds, int *strides, int *offsets,
+               void *loaderPrivate)
+ {
++   uint64_t modifier= DRM_FORMAT_MOD_INVALID;
++   unsigned int external_only= 0;
++   int count= 0;
++   boolean result;
++   result= dri2_query_dma_buf_modifiers( screen, fourcc, 1, &modifier, &external_only, &count);
++   printf("====JRW: query result %d count %d external_only %d modifier %llx\n", result, count, external_only, modifier);
```
Comment (kraj) [inline recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch]: get rid of this warning

File: recipes-graphics/mesa/mesa_%.bbappend [hunk 2]
```diff
@@ -2,5 +2,9 @@
 # With oe-core commit 8509e2e1a87578882b71948ccef3b50ccf1228b3 dri3 is set
 # as default. To state out clearly that Raspi needs dri3 and to avoid surprises
 # in case oe-core changes this default, we set dri3 explicitly.
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+SRC_URI += " file://0001-dri2-query-dma-buf-modifiers.patch "
```
Comment (kraj) [inline recipes-graphics/mesa/mesa_%.bbappend]: call it `SRC_URI_append_rpi = ...`

**Comment by balav08** [comment]
Hi @kraj we have updated the changes please review.

File: recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch [hunk 3]
```diff
@@ -0,0 +1,33 @@
+diff --git a/src/gallium/state_trackers/dri/dri2.c b/src/gallium/state_trackers/dri/dri2.c
```
Comment (agherzan) [inline recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch]: Add upstream status - see other patches for reference. https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines

Comment (agherzan) [reply recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch]: Can we also have a description of the patch in both the patch file and git log?

**Comment by balav08** [comment]
Hi @agherzan we have updated the description in patch and git log please review.

File: recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch (hunk 3 above)
Comment (agherzan) [reply recipes-graphics/mesa/files/0001-dri2-query-dma-buf-modifiers.patch]: - Upstream status was forgotten.
- SOB in commit log (capital letter for "signed")
- Please use your name before the email - see other commits as an example and/or use `-s` argument of git commit (and co).

File: recipes-graphics/mesa/mesa_%.bbappend [hunk 4]
```diff
@@ -2,5 +2,9 @@
 # With oe-core commit 8509e2e1a87578882b71948ccef3b50ccf1228b3 dri3 is set
 # as default. To state out clearly that Raspi needs dri3 and to avoid surprises
 # in case oe-core changes this default, we set dri3 explicitly.
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (agherzan) [inline recipes-graphics/mesa/mesa_%.bbappend]: You should add this to the top of the file. Inserting after the comment above adds confusion.

**Comment by agherzan** [comment]
@agherzan-ci ok to test

**Comment by agherzan** [comment]
@kraj small ping for a review here when you get a chance.

**Comment by agherzan** [comment]
@balav08 Can you please test it on mesa 20?

**Comment by balav08** [comment]
Hi @agherzan  we have tested these changes with mesa 20. And its working fine.

**Comment by agherzan** [comment]
Merged. Thanks, @balav08 .

**Pull request merged**